iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/HiveIcebergRESTCatalogClientAdapter.java

```diff
@@ -0,0 +1,2426 @@
+    storageDescriptor.setInputFormat(DEFAULT_INPUT_FORMAT_CLASS);
+    storageDescriptor.setOutputFormat(DEFAULT_OUTPUT_FORMAT_CLASS);
+    storageDescriptor.setBucketCols(new LinkedList<>());
+    storageDescriptor.setSortCols(new LinkedList<>());
+    storageDescriptor.setParameters(Maps.newHashMap());
+    if (icebergTable.properties().containsKey(SERDE_INFO)) {
+      // TODO
+    } else {
+      SerDeInfo serDeInfo = new SerDeInfo("icebergSerde", DEFAULT_SERDE_CLASS, Maps.newHashMap());
+      serDeInfo.getParameters().put(serdeConstants.SERIALIZATION_FORMAT, "1"); // Default serialization format.
+      storageDescriptor.setSerdeInfo(serDeInfo);
+    }
+    icebergTable.schema().columns().forEach(icebergColumn -> {
+      FieldSchema fieldSchema = new FieldSchema();
+      fieldSchema.setName(icebergColumn.name());
+      fieldSchema.setType(icebergColumn.type().toString());
+      cols.add(fieldSchema);
+    });
+    return hiveTable;
+  }
+  private String getTableName(org.apache.iceberg.Table icebergTable) {
+    String[] nameParts = icebergTable.name().split("\\.");
+    if (nameParts.length == 3) {
+      return nameParts[2];
+    }
+    if (nameParts.length == 2) {
+      return nameParts[1];
+    }
+    return icebergTable.name();
+  }
+
+  private String getDbName(org.apache.iceberg.Table icebergTable) {
+    String[] nameParts = icebergTable.name().split("\\.");
+    return nameParts.length == 3 ? nameParts[1] : nameParts[0];
+  }
+
+  @Override
+  public Table getTable(String dbName, String tableName, boolean getColumnStats, String engine)
+      throws MetaException, TException, NoSuchObjectException {
+    return getTable(dbName, tableName);
+  }
+
+  @Override
+  public Table getTable(String catName, String dbName, String tableName) throws MetaException, TException {
+    return getTable(dbName, tableName);
+  }
+
+  @Override
+  public Table getTable(String catName, String dbName, String tableName, String validWriteIdList) throws TException {
+    return getTable(dbName, tableName);
+  }
+
+  @Override
+  public Table getTable(String catName, String dbName, String tableName, String validWriteIdList,
+      boolean getColumnStats, String engine) throws TException {
+    return getTable(dbName, tableName);
+  }
+
+  @Override
+  public Table getTable(GetTableRequest getTableRequest) throws MetaException, TException, NoSuchObjectException {
+    return getTable(getTableRequest.getDbName(), getTableRequest.getTblName());
+  }
+
+  @Override
+  public List<Table> getTableObjectsByName(String dbName, List<String> tableNames)
+      throws MetaException, InvalidOperationException, UnknownDBException, TException {
+    return null;
+  }
+
+  @Override
+  public List<Table> getTables(String catName, String dbName, List<String> tableNames,
+      GetProjectionsSpec projectionsSpec)
+      throws MetaException, InvalidOperationException, UnknownDBException, TException {
+    return new LinkedList<>();
```

Comment:
I believe returning a new empty list via instantiation is not the best practice in Java, should we change to 
`Collections.emptyList()` or `List.of();` (Java 9+ syntax, we are now on Java 17).

Reply:
Refactored in IMetaStoreClient.